mkosi/__init__.py

```diff
@@ -266,32 +266,29 @@ def run_prepare_script(state: MkosiState, build: bool) -> None:
         BUILDROOT=str(state.root),
     )
 
-    chroot: list[PathString] = chroot_cmd(
-        state.root,
-        options=[
-            "--bind", state.config.prepare_script, "/work/prepare",
-            "--bind", Path.cwd(), "/work/src",
-            "--chdir", "/work/src",
-            "--setenv", "SRCDIR", "/work/src",
-            "--setenv", "BUILDROOT", "/",
-        ],
-    )
-
-    if build:
-        with complete_step("Running prepare script in build overlay…"), mount_build_overlay(state):
-            bwrap(
-                [state.config.prepare_script, "build"],
-                network=True,
-                readonly=True,
-                options=finalize_mounts(state.config),
-                scripts={"mkosi-chroot": chroot} | package_manager_scripts(state),
-                env=env | state.config.environment,
-                stdin=sys.stdin,
```

Comment:
These seem to be gone?

Reply:
They are not gone. There was only one call to `bwrap` for the script, with a few tweaks on the parameters depending on if `build` is set or not and the call to `mount_build_overlay`.

I extracted the variations of `if build: … else: …` in a setup phase and only kept a single call to bwrap for each script.

Reply:
Right but according to github's UI bwrap is now being called as follows:

```
bwrap(
                [script, arg],
                network=True,
                readonly=True,
                options=finalize_mounts(state.config),
	@@ -301,8 +298,8 @@ def run_prepare_script(state: MkosiState, build: bool) -> None:
            )
```

Which is missing the `env` and `stdin` arguments that were there before

Reply:
They re not missing, they are just collapsed in the diff and they appear if you expand the `@@`.